test/test_module_init.py

```diff
@@ -0,0 +1,330 @@
+    MODULES_WITHOUT_KWARGS_SUPPORT = {
+        torch.nn.BCELoss,
+        torch.nn.BCEWithLogitsLoss,
+        torch.nn.CrossEntropyLoss,
+        torch.nn.FractionalMaxPool2d,
+        torch.nn.FractionalMaxPool3d,
+        torch.nn.MultiLabelSoftMarginLoss,
+        torch.nn.MultiMarginLoss,
+        torch.nn.NLLLoss,
+        torch.nn.TransformerDecoder,
+        torch.nn.TransformerEncoder,
+    }
+    # modules that supported kwargs before
+    MODULES_WITH_PREVIOUS_KWARGS = {
+        torch.nn.Identity,
+    }
+    # lazy modules don't instantiate parameters right away
+    LAZY_MODULES = {
+        torch.nn.LazyBatchNorm1d,
+        torch.nn.LazyBatchNorm2d,
+        torch.nn.LazyBatchNorm3d,
+        torch.nn.LazyConv1d,
+        torch.nn.LazyConv2d,
+        torch.nn.LazyConv3d,
+        torch.nn.LazyConvTranspose1d,
+        torch.nn.LazyConvTranspose2d,
+        torch.nn.LazyConvTranspose3d,
+        torch.nn.LazyConvTranspose3d,
+        torch.nn.LazyLinear,
+    }
+
+    for namespace in NAMESPACES:
+        # the "nn" in "torch.nn"
+        namespace_basename = namespace.__name__.split('.')[-1]
+        # namespace object (e.g. torch.nn.modules)
+        mod_namespace = getattr(namespace, 'modules')
+        # all module names (e.g. "Linear") in the namespace
+        module_names = getattr(mod_namespace, '__all__')
+        for module_name in module_names:
+            # class object for this module (e.g. torch.nn.Linear)
+            module_cls = getattr(mod_namespace, module_name)
+            if module_cls in MODULES_TO_SKIP: continue
+
+            # Create a function to run the test and setattr it onto the test class.
+            def run_test(test_cls, device, namespace_basename=namespace_basename,
+                         module_name=module_name, module_cls=module_cls):
+                # Check if this module creates parameters or registers buffers.
+                # The mock magic here passes through to the real Parameter / register_buffer
+                # logic and is only used to check for calls.
+                args, kwargs = get_example_args(module_cls, constructor_arg_db, device=None)
+                parameter_new = mock_wrapper(torch.nn.Parameter.__new__)
+                with patch.object(torch.nn.Parameter, '__new__', parameter_new):
+                    register_buffer = mock_wrapper(torch.nn.Module.register_buffer)
+                    with patch.object(torch.nn.Module, 'register_buffer', register_buffer):
+                        m = module_cls(*args, **kwargs)
+                        module_creates_params_or_buffers = parameter_new.mock.called or register_buffer.mock.called
+
+                # == Verify kwargs logic. ==
+                if module_creates_params_or_buffers and module_cls not in MODULES_WITHOUT_KWARGS_SUPPORT:
+                    args, kwargs = get_example_args(module_cls, constructor_arg_db, device=device)
+
+                    # if module_cls in LAZY_MODULES:
```

Comment:
What's up with this comment?

Reply:
This is testing logic for LazyTensors that should be put back in once meta tensor functionality is expanded. Not sure whether this PR should wait for that so the tests will be better?

Reply:
I probably wouldn't wait (although you know better). I would just remove this comment (possibly replacing it with a TODO) or add a meta-comment explaining why this section is commented out.

Reply:
Luckily enough, the support is in now so I uncommented it!